index.bs

```diff
@@ -426,19 +426,25 @@ particularly important for the data structure described in [[#queue-with-sizes]]
  const reader = readableStream.getReader({ mode: "byob" });
 
  let startingAB = new ArrayBuffer(1024);
- readInto(startingAB)
-   .then(buffer => console.log("The first 1024 bytes:", buffer))
-   .catch(e => console.error("Something went wrong!", e));
-
- function readInto(buffer, offset = 0) {
-   if (offset === buffer.byteLength) {
-     return Promise.resolve(buffer);
-   }
+ const buffer = await readInto(startingAB);
+ console.log("The first 1024 bytes: ", buffer);
+
+ async function readInto(buffer) {
+   let offset = 0;
+   let view;
+   let done;
+
+   do {
```

Comment:
I *think* this could be a regular `while` loop? If `buffer.byteLength === 0`, then we don't even need to bother calling `read()`.
```suggestion
   while (offset < buffer.byteLength) {
```

Reply:
Agree. A regular `while` loop is easier to understand.

Reply:
That's a good point! I've just changed it to that in the last commit. Thank you!